make_mt_IHME.py

```diff
@@ -0,0 +1,254 @@
+
+This can be imported to use the following functions:
+convert_ihme_mt_to_table(mt: hl.MatrixTable, age_groups: list, explode_icd: str) -> hl.Table
+
+@author: rgupta
+"""
+import hail as hl
+import numpy as np
+from copy import deepcopy
+import argparse, re
+__author__ = 'rgupta'
+
+
+def get_entry_cols(entry_substr: str, field_names: list):
+    entry_substr_l = entry_substr.split(',')
+    matching_list_entries = []
+    for ele in entry_substr_l:
+        matching_items = [field for field in field_names if re.search(ele,field)]
+        matching_list_entries = matching_list_entries + matching_items
+    return list(set(matching_list_entries))
+
+
+def get_column_cols(col_key: str, col_other: str, field_names: list):
+    col_other_l = col_other.split(',')
+    if not all([col in field_names for col in col_other_l]):
+        raise ValueError('All column fields must be in the columns of the read table.')
+    if not col_key in field_names:
+        raise ValueError('Column key must be in set of fields.')
+    return col_key, list(np.setdiff1d(list(set(col_other_l)),[col_key]))
+
+
+def get_row_cols(row_key: str, row_other: list, col_fields: list, 
+                 entry_fields: list, field_names: list):
+    if not row_key in field_names:
+        raise ValueError('Row key must be in set of fields.')
+    if(len(row_other) == 0):
+        row_other_l = np.setdiff1d(np.setdiff1d(field_names, col_fields), entry_fields)
+    else:
+        row_other_l = row_other.split(',')
+        if not all([row in field_names for row in row_other_l]):
+            raise ValueError('All row fields must be in the columns of the read table.')
+    return row_key, list(np.setdiff1d(list(set(row_other_l)),[row_key]))
+
+
+def recurse_struct(string_array: list, depth: int, ordering: list, 
+                   elements: list, ht: hl.Table):
+    """A recursive function to produce a nested struct
+
+    Parameters
+    ----------
+    string_array : list
+        An array that has the same length as the depth of the desired nested structure. Should contain
+        None for any positions that haven't been visited yet. Once this array is filled (i.e.,
+        at the deepest position), will be concatenated into a period-delimited string which should
+        be found in the Hail Table.
+    
+    depth : int
+        The depth of the strucure to currently iterate in.
+    
+    ordering : list
+        This is used to map the elements of the column names (period delimited) to
+        the struct. This function reverses this order, taking elements (which is 
+        ordered in the order of the nested structure) and converting it back to the
+        order of the original column name.
+    
+    elements : list
+        A list of lists. Each element forms the names of that depth of the nested struct.
+    
+    ht : Hail Table
+        Should contain columns that will be transformed into the new struct.
+
```

Comment:
Can you provide a simple pictoral example of what this function does? Hard to envision from just the docstring